### PR TITLE
Fix disruptive tests for GKE regional clusters

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -4095,6 +4095,18 @@ func CheckForControllerManagerHealthy(duration time.Duration) error {
 	return nil
 }
 
+// Returns number of registered Nodes excluding Master Node.
+func NumberOfNodes(c clientset.Interface) (int, error) {
```
Comment (shyamjvs) [inline test/e2e/framework/util.go]: I think there should be a function like this somewhere already IIRC. And that one should also have retry-logic. Can you please check for it?

Comment (wojtek-t) [reply test/e2e/framework/util.go]: I was looking for it and there isn't.

About retries - we should probably fix it, but for now it's consistent with a number of functions underneath doing similar stuff. We should fix the all together.

Comment (shyamjvs) [reply test/e2e/framework/util.go]: You can probably re-use `waitListSchedulableNodesOrDie()`.

Comment (shyamjvs) [reply test/e2e/framework/util.go]: See above ^

Comment (wojtek-t) [reply test/e2e/framework/util.go]: Good finding. Done.

Comment (shyamjvs) [reply test/e2e/framework/util.go]: Also maybe rename this function to `NumberOfNodesRegistered` or sth - to make it clearer what it's giving?

**Pull request merged**